code/search/SearchCriterion.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+class SearchCriterion implements SearchCriteriaInterface
+{
+    /**
+     * field:value
+     *
+     * @var string
+     */
+    const EQUAL = 'EQUAL';
+
+    /**
+     * -field:value
+     *
+     * @var string
+     */
+    const NOT_EQUAL = 'NOT_EQUAL';
+
+    /**
+     * field:[value TO *]
+     *
+     * @var string
+     */
+    const GREATER_EQUAL = 'GREATER_EQUAL';
+
+    /**
+     * field:{value TO *}
+     *
+     * @var string
+     */
+    const GREATER_THAN = 'GREATER_THAN';
+
+    /**
+     * field:[* TO value]
+     *
+     * @var string
+     */
+    const LESS_EQUAL = 'LESS_EQUAL';
+
+    /**
+     * field:{* TO value}
+     *
+     * @var string
+     */
+    const LESS_THAN = 'LESS_THAN';
+
+    /**
+     * (field:value1 field:value2 field:value3)
+     *
+     * @var string
+     */
+    const IN = 'IN';
+
+    /**
+     * -(field:value1 field:value2 field:value3)
+     *
+     * @var string
+     */
+    const NOT_IN = 'NOT_IN';
+
+    /**
+     * field:[* TO *]
+     *
+     * @var string
+     */
+    const ISNULL = 'ISNULL';
+
+    /**
+     * -field:[* TO *]
+     *
+     * @var string
+     */
+    const ISNOTNULL = 'ISNOTNULL';
+
+    /**
+     * A custom Criterion with it's own SearchQueryWriter
+     *
+     * @var string
+     */
+    const CUSTOM = 'CUSTOM';
+
+    /**
+     * @var string
+     */
+    protected $comparison;
+
+    /**
+     * The table and field that this Criterion is applied to.
+     *
+     * @var string
+     */
+    protected $target;
+
+    /**
+     * @var mixed
+     */
+    protected $value;
+
+    /**
+     * @var SearchAdapterInterface
+     */
+    protected $adapter;
+
+    /**
+     * @var AbstractSearchQueryWriter
+     */
+    protected $searchQueryWriter;
+
+    /**
+     * @param string $target
+     * @param mixed $value
+     * @param string|null $comparison
+     * @param AbstractSearchQueryWriter $searchQueryWriter
+     */
+    public function __construct(
+        $target,
+        $value,
+        $comparison = null,
+        AbstractSearchQueryWriter $searchQueryWriter = null
+    ) {
+        // EQUAL is our default comparison.
+        if ($comparison === null) {
+            $comparison = SearchCriterion::EQUAL;
+        }
+
+        $this->setTarget($target);
+        $this->setValue($value);
+        $this->setComparison($comparison);
+        $this->setSearchQueryWriter($searchQueryWriter);
+    }
+
+    /**
+     * Static create method provided so that you can implement method chaining.
+     *
+     * @return SearchCriterion
+     */
+    public static function create()
+    {
+        $args = func_get_args();
+
+        $class = get_called_class();
+        if ($class == 'Object') {
+            $class = array_shift($args);
+        }
+
+        return Injector::inst()->createWithArgs($class, $args);
```

Comment:
Is there a reason we can't just use `::create()` directly from Object/injector for this over defining our own `create` methods?